ql/pricingengines/swap/treeswapengine.cpp

```diff
@@ -71,7 +71,7 @@ namespace QuantLib {
             lattice = model_->tree(timeGrid);
         }
 
-        swap.initialize(lattice, times.back());
+        swap.initialize(lattice, lattice->timeGrid().back());
```

Comment:
I'm afraid using the last time on the lattice could be equally misleading; for some instruments, like swaptions, the lattice extends beyond the time at which the instrument needs to be initialized.  All in all, I'd prefer the initialization to refer explicitly to the time we want.  If we wanted to protect against unsorted times, we could use `max_element` instead of `back` to retrieve the latest one.  What do you think?

Reply:
Like this?
```cpp
        Time maxTime = *std::max_element(times.begin(), times.end());
        swap.initialize(lattice, maxTime);
```
Fine with me. I will update my PR.